wallet/src/account/output_cache/mod.rs

```diff
@@ -1401,90 +1449,157 @@ impl OutputCache {
         })
     }
 
-    /// Mark a transaction and its descendants as abandoned
-    /// Returns a Vec of the transaction Ids that have been abandoned
-    pub fn abandon_transaction(
+    // Removes a tx and all its descendants from unconfirmed descendants collection.
+    // Returns provided tx and all the descendants in such way that parent always comes before its
+    // descendants. Returned collection contains no duplicates.
+    fn remove_from_unconfirmed_descendants(
         &mut self,
         tx_id: Id<Transaction>,
-    ) -> WalletResult<Vec<Id<Transaction>>> {
-        let mut all_abandoned = Vec::new();
-        let mut to_abandon = BTreeSet::from_iter([OutPointSourceId::from(tx_id)]);
-
-        while let Some(outpoint_source_id) = to_abandon.pop_first() {
-            all_abandoned.push(*outpoint_source_id.get_tx_id().expect("must be a transaction"));
+    ) -> Vec<Id<Transaction>> {
+        let mut all_txs = Vec::new();
+        let mut all_txs_as_set = BTreeSet::new();
+        let mut to_update: VecDeque<OutPointSourceId> = [tx_id.into()].into();
+
+        while let Some(outpoint_source_id) = to_update.pop_front() {
+            let tx_id = *outpoint_source_id.get_tx_id().expect("must be a transaction");
+            if all_txs_as_set.insert(tx_id) {
+                all_txs.push(tx_id);
+            }
 
             if let Some(descendants) = self.unconfirmed_descendants.remove(&outpoint_source_id) {
-                to_abandon.extend(descendants.into_iter())
+                to_update.extend(descendants.into_iter())
             }
         }
 
+        all_txs
+    }
+
+    // After tx is removed as a result of reorg, abandoning or marking as conflicted
+    // its effect on OutputCache's data fields should be rolled back
+    fn rollback_tx_data(&mut self, tx: &WalletTx) -> WalletResult<()> {
+        let tx_id = tx.id();
+
+        // Iterate in reverse to handle situations where an account is modified twice in the same tx
```

Comment:
[discussed] Mention that this situation is not currently possible due to the restriction in the consensus that only one account command is permitted per tx.